Adding Fletcher et al. (2017) data

drisso opened the pull request:
Hi @LTLA ,

once you confirm that everything looks good I will upload the rds files to S3.

This is untested (and I didn't add unit tests) until the files are in ExperimentHub.

Let me know how to proceed, e.g., upload the files to the hub -> add tests to this PR -> merge?

LTLA commented:
It looks good. I would rename to `FletcherOlfactoryData` and `fletcher-olfactory`. Also the BiocVersion should be the next one, 3.13.

LTLA commented on R/FletcherOEData.R:
```diff
@@ -0,0 +1,69 @@
+#'   be returned.
+#' @details
+#' Column metadata is scraped from GEO, using both the author-supplied
+#' "phenoData" per-cell annotations and the author-supplied "protocolData"
```
Use `\dQuote`.

drisso replied:
Done

LTLA commented on R/FletcherOEData.R:
```diff
@@ -0,0 +1,69 @@
+#'
+#' If \code{filtered=TRUE}, only the cells used by the authors in their cluster
+#' analysis are returned. Otherwise, the cells not used by the authors will have
+#' NA in the clustering columns of the \code{\link{colData}}.
```
`\code{NA}`.

drisso replied:
Done

LTLA commented on R/FletcherOEData.R:
```diff
@@ -0,0 +1,69 @@
+    sce <- .create_sce(file.path("fletcher-oe", version), has.rowdata=TRUE)
+
+    if (filtered) {
+        sce <- sce[,!is.na(sce$cluster_id)]
```
I would prefer we had a `retained` TRUE/FALSE column for consistency with others, e.g., `JessaBrainData`. In principle, a `NA` field in the clustering could mean other things, e.g., the cell was retained but the authors didn't know what it was.

drisso replied:
I'm not sure that I understand, doens't `JessaBrainData` also have a `filtered=TRUE` argument?

LTLA replied:
Yes, and it also has an explicit `retained` column indicating whether that cell was retained.

drisso replied:
Oh I see what you mean now, I'll add the column

LTLA commented:
Also need to add it in `inst/extdata/manifest.csv`.

drisso commented:
Ok, I fixed the minor documentation things, renamed to `fletcher-olfactory` and `FletcherOlfactoryData` and updated the Bioc version.

If you have your EHub credentials ready, I don't mind if you compile and upload.

drisso commented:
@LTLA do you want me to create a new PR for the `retained` column or did you already push the files to EHub?

LTLA commented:
I already did it and pushed it. When it becomes available, we'll probably need to patch up the getter.
